src/App/Search/AbstractSearchClient.php

```diff
@@ -346,4 +347,124 @@ protected function checkKeywordAgainstSynonyms(string $searchPhrase): string
 
         return $searchPhrase;
     }
+
+    /**
+     * reindexes search index
+     *
+     * @param string $indexName
+     * @param array $analysis
+     *
+     */
+
+    public function reindex(string $indexName, array $analysis)
+    {
+        // call parent constructer so we can get access to elastica client methods
+        parent::__construct();
+
+        $oldIndex = $this->getIndex($indexName);
+        $newIndex = $this->getIndex($indexName . '_temp');
+
+        // create new temp index with new index settings to copy documents to
+        $this->createNewIndex($indexName . '_temp', $analysis);
+        
+        // copy documents from old index to new index
+        $reindexAPI = new Reindex($oldIndex, $newIndex);
+        $reindexAPI->run();
+
+        // delete old index
+        $oldIndex->delete();
+
+        // create new index with same name as old index
+        $this->createNewIndex($indexName, $analysis);
+
+        // copy documents from new temp index to old index(with the new data and settings)
+        $reindexAPI = new Reindex($newIndex, $oldIndex);
+        $reindexAPI->run();
+        
+        // delete new temp index
+        $newIndex->delete();
+    }
+
+     /**
+     * creates a new index and adds new analysis
+     *
+     * @param string $indexName
+     * @param array $analysis
+     *
+     */
+
+    public function createNewIndex(string $indexName, array $analysis)
+    {
+        $index = $this->getIndex($indexName);
+
+        $index->create(['settings' => $analysis]);
+
+        $index->setMapping($this->getIndexMapping());
+    }
+
+    /**
+     * overrides current index analysis for framework
+     */
+    public function overrideFrameworkIndex()
+    {
+    // update index setting here
+        $analysis = [
+        'analysis' => array(
+          'analyzer' => array(
+            'english_analyzer' => array(
+              'tokenizer' => 'standard',
+              'filter'    => array('lowercase', 'english_stemmer', 'english_stop'),
+            ),
+          ),
+          'filter'   => array(
+            'english_stemmer' => array(
+              'type' => 'stemmer',
+              'name' => 'english'
+            ),
+            'english_stop' => array(
+              'type' => 'stop',
+              'stopwords' => '_english_'
+            )
+          )
+        )
+        ];
+
+        // create old index if it doesn't exist
+        $this->getIndexOrCreate();
+
+        $this->reindex($this->getQualifiedIndexName(), $analysis);
+    }
+
+    /**
+     * overrides current index analysis for supplier
+     */
+    public function overrideSupplierIndex()
```

Comment:
No, you have 2 identical functions with different names. You only need one function with a generic name without the word _override_. Please read about [function overriding ](https://www.w3schools.com/php/php_oop_inheritance.asp). 
If we wanted to add changes to the Framework analyser at a later date, we could override the function in FrameworkSearchClient